include/pybind11/cast.h

```diff
@@ -2007,7 +2041,7 @@ class argument_loader {
 
     template <typename Return, typename Func, size_t... Is, typename Guard>
     Return call_impl(Func &&f, index_sequence<Is...>, Guard &&) && {
-        return std::forward<Func>(f)(cast_op<Args>(std::move(std::get<Is>(argcasters)))...);
+        return std::forward<Func>(f)(cast_op<Args>(std::get<Is>(argcasters))...);
```

Comment:
This change is crucial: Not moving the argcasters here, avoids adding an rvalue reference to all cast_op targets.